Arch tidy of mix concerns (#963)
ietf-tapswg/api-drafts: draft-ietf-taps-arch.md

By default, stored properties of the implementation, such as cached protocol state, cached path state, and heuristics, may be shared (e.g. across multiple connections in an application). This provides efficiency and convenience for the application, since the Transport Services system can automatically optimize behavior.

The Transport Services API can allow applications to explicitly define Connection Contexts that force separation of Cached State and Protocol Stacks. 

> For example, a web browser a web browser application could use Connection Contexts with separate caches when implementing different tabs. Possible reasons to isolate Connections using separate Connection Contexts include:

Reviewer: ```suggestion
For example, a web browser application could use Connection Contexts with separate caches when implementing different tabs. Possible reasons to isolate Connections using separate Connection Contexts include:
```

> - Privacy concerns about re-using cached protocol state that can lead to linkability. Sensitive state could include TLS session state {{?RFC8446}} and HTTP cookies {{?RFC6265}}. A web browser application could use Connection Contexts with separate caches when implementing different tabs. 

Author: Should now be resolved following your text

> - Privacy concerns about re-using cached protocol state that can lead to linkability. Sensitive state could include TLS session state {{?RFC8446}} and HTTP cookies {{?RFC6265}}. This could use Connection Contexts with separate caches when implementing different browser tabs. 

Reviewer 2: ```suggestion
- Privacy concerns about re-using cached protocol state that can lead to linkability. Sensitive state could include TLS session state {{?RFC8446}} and HTTP cookies {{?RFC6265}}. These concerns could be addressed using Connection Contexts with separate caches, such as for different browser tabs. 
```
